### PR TITLE
Adding Uber's hermetic cc toolchain for tests

### PR DESCRIPTION
**Pull request by linzhp**
**What type of PR is this?**
Other

**What does this PR do? Why is it needed?**
Enable [uber/hermetic_cc_toolchain](https://github.com/uber/hermetic_cc_toolchain) to run tests on CI

**Other notes for review**
@fmeum Is there a better way to detect the current C toolchain in the test, so I can skip some test cases?

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -179,3 +179,24 @@ load(
...
 
 apple_support_dependencies()
+
+# For testing the compatibility with a hermetic cc toolchain. Users should not have to enable it.
+
+http_archive(
+    name = "hermetic_cc_toolchain",
+    sha256 = "bd2234acd0837251361be3270d7d3ce599b418be123d902d84762302e31a3014",
+    strip_prefix = "hermetic_cc_toolchain-13c904dce0cb9b6d07f0d557e6ce3cf7013a562e",
+    urls = ["https://github.com/uber/hermetic_cc_toolchain/archive/13c904dce0cb9b6d07f0d557e6ce3cf7013a562e.zip"],
```
Comment (motiejus) [inline WORKSPACE:189]: curious, why not the released version? Any known blockers?

File: .bazelci/presubmit.yml [hunk 2]
```diff
@@ -12,21 +12,37 @@ tasks:
...
-    - "--config=incompatible"
+      - "--config=incompatible"
+    test_flags:
+      - "--config=incompatible"
+    build_targets:
+      - "//..."
+    test_targets:
+      - "//..."
+  ubuntu2004_uber_hermetic_cc:
+    platform: ubuntu2004
```
Comment (motiejus) [inline .bazelci/presubmit.yml:31]: not sure where to get the list of open platforms, but is there one that doesn't have a C/C++ compiler installed?

**Review by motiejus — COMMENTED** [review]
It would be great to split `.bazelci/presubmit.yaml` lint checks to a separate commit or even PR for easier review.

Also, please squash the changes, there are extra unneeded merges.

File: WORKSPACE (hunk 1 above)
Comment (linzhp) [reply WORKSPACE:189]: Yeah, I need https://github.com/uber/hermetic_cc_toolchain/commit/a9d87b21a5dddd691336c6c0004fa5dcfe5b9b48 to fix some tests.

File: .bazelci/presubmit.yml (hunk 2 above)
Comment (linzhp) [reply .bazelci/presubmit.yml:31]: This is the list: https://github.com/bazelbuild/continuous-integration/blob/8c5e4b1322ea5751842bfd03d2e03d31dbe1f8f2/buildkite/bazelci.py#L465-L615

I am planning to add macOS cross-compiling into Linux, which is a common scenario

**Comment by linzhp** [comment]
> It would be great to split `.bazelci/presubmit.yaml` lint checks to a separate commit or even PR for easier review.
Split to #3611 

> Also, please squash the changes, there are extra unneeded merges.
Will be squashed before merge

File: .bazelrc [hunk 3]
```diff
@@ -1,11 +1,17 @@
 common --enable_platform_specific_config
+test --test_output=errors
 
 # Go requires a C toolchain that accepts options and emits errors like
+
```
Comment (fmeum) [inline .bazelrc:5]:
```suggestion
```

**Review by fmeum — APPROVED** [review]

**Pull request merged**